front/components/assistant_builder/actions/ProcessAction.tsx

```diff
@@ -254,34 +254,41 @@ function PropertiesFields({
   );
 }
 
-export function ActionProcess({
-  owner,
-  instructions,
-  actionConfiguration,
-  updateAction,
-  setEdited,
-  description,
-  onDescriptionChange,
-}: {
+type ActionProcessProps = {
   owner: WorkspaceType;
   instructions: string | null;
   actionConfiguration: AssistantBuilderProcessConfiguration | null;
+  allowedVaults: VaultType[];
```

Comment:
Given that we need to spread vaults in many places now, I'm wondering if we don't want to leverage the existing assistant builder context to avoid adding all those new props. Context can have state management function so we could think of a function to get the allowed vaults based on the current action, WDYT?

Reply:
I'm not a fan, it make the "logic" spread between the global and local context. I think it will make it harder to follow the code flow.